src/main/scala/org/dia/algorithms/mcc/GTGRunner.scala

```diff
@@ -336,6 +336,15 @@ class GTGRunner(val masterURL: String,
     processEdges(MCCEdgeList, MCCNodeMap)
 
     /**
+     * Generate the netcdfs
+     */
+    val broadcastedNodeMap = sc.sparkContext.broadcast(MCCNodeMap)
+    val edgeTupleList = sc.sparkContext.parallelize(MCCEdgeList.toList).map({ x =>
+      (s"${x.srcNode.frameNum},${x.srcNode.cloudElemNum}",
+      s"${x.destNode.frameNum},${x.destNode.cloudElemNum}") })
+    MCSUtils.get_all_node_data(edgeTupleList.collect().toArray, broadcastedNodeMap.value, lat, lon, false)
```

Comment:
There is a redundant toArray here.
Collect by default returns you an array of the elements in an RDD.
An extra toArray will have no effect.

Take a look at the return type of collect in the scaladoc:
http://spark.apache.org/docs/latest/api/scala/index.html#org.apache.spark.rdd.RDD@collect():Array[T]